tools/xml2cpp-codegen/ProxyGenerator.cpp

```diff
@@ -150,6 +152,12 @@ std::tuple<std::string, std::string> ProxyGenerator::processMethods(const Nodes&
             else if (annotation->get("name") == "org.freedesktop.DBus.Method.Async"
                      && (annotation->get("value") == "client" || annotation->get("value") == "clientserver"))
                 async = true;
+
+           if ((annotation->get("name") == "org.freedesktop.DBus.Method.Async") && (annotation->get("value") == "client")
```

Comment:
We don't need this `if` at all if we provide timeout support also for sync calls (which we should now that we are adding timeout for async ones). We only need `timeoutValue = annotation->get("timeout");`, and later in the code we just check whether timeoutValue is non-empty string. If it is, we simply add `.withTimeout(timeoutValue)` to the call invocation command.